android-components/components/feature/media/src/main/java/mozilla/components/feature/media/fullscreen/MediaSessionFullscreenFeature.kt

```diff
@@ -35,51 +35,53 @@ class MediaSessionFullscreenFeature(
+                    old?.content?.pictureInPictureEnabled == new?.content?.pictureInPictureEnabled &&
+                    old?.mediaSessionState?.elementMetadata == new?.mediaSessionState?.elementMetadata
+            }.collect { state -> /* there should only be one fullscreen session */
+                if (state == null || state.mediaSessionState?.fullscreen != true) {
+                    activity.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_USER
+                    activity.window?.clearFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON)
+                    return@collect
+                }
+
+                if (store.state.findCustomTabOrSelectedTab(tabId)?.id == state.id) {
+                    processFullscreen(state)
+                }
+                processDeviceSleepMode(state)
             }
         }
     }
 
     @Suppress("SourceLockedOrientationActivity") // We deliberately want to lock the orientation here.
-    private fun processFullscreen(sessionStates: List<SessionState>) {
-        /* there should only be one fullscreen session */
-        val activeState = sessionStates.firstOrNull()
-        if (activeState == null || activeState.mediaSessionState?.fullscreen != true) {
-            activity.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_USER
-            return
-        }
+    private fun processFullscreen(activeTabState: SessionState) {
+        when (activeTabState.mediaSessionState?.elementMetadata?.portrait) {
+            true ->
+                activity.requestedOrientation =
+                    ActivityInfo.SCREEN_ORIENTATION_USER_PORTRAIT
 
-        if (store.state.findCustomTabOrSelectedTab(tabId)?.id == activeState.id) {
-            when (activeState.mediaSessionState?.elementMetadata?.portrait) {
-                true ->
+            false ->
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N && activity.isInPictureInPictureMode) {
+                    activity.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_UNSPECIFIED
+                } else {
                     activity.requestedOrientation =
-                        ActivityInfo.SCREEN_ORIENTATION_USER_PORTRAIT
-                false ->
-                    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N && activity.isInPictureInPictureMode) {
-                        activity.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_UNSPECIFIED
-                    } else {
-                        activity.requestedOrientation =
-                            ActivityInfo.SCREEN_ORIENTATION_SENSOR_LANDSCAPE
-                    }
-                else -> activity.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_USER
-            }
+                        ActivityInfo.SCREEN_ORIENTATION_SENSOR_LANDSCAPE
+                }
+
+            else -> activity.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_USER
```

Comment:
nit: while we're changing this anyway, I personally would find `null` more clear than `else`. The `else` gives me a bit of anxiety wondering what other state a boolean could possibly be in besides `true` or `false` 😂